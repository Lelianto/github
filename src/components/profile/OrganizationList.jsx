import React from 'react';
import { Link } from 'react-router-dom';

const OrganizationList = ({ organizations }) => {
	return (
		<>
			{
				organizations.length && organizations.length !== 0 ?
					<>
						{
							organizations.map((organization) => {
								return (
									<Link key={organization.id} to={`/org/${organization.login}`}>
										<img className="mr-1 mb-1" size="32px" height="32px" width="32px" src={organization.avatar_url} alt="" />
									</Link>
								)
							})
						}
					</>
					:
					<></>
			}
		</>
	)
}

export default OrganizationList