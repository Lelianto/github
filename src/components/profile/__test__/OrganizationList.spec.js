import OrganizationList from "../OrganizationList";
import { shallow } from "enzyme";
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';

describe("Organization List: Check content", () => {
	let wrapper;
	let mockProps = [
		{
			id: 123,
			login: 'Organization1',
			avatar_url: 'https://upload.wikimedia.org/wikipedia/commons/9/91/Octicons-mark-github.svg'
		},
		{
			id: 234,
			login: 'Organization2',
			avatar_url: 'https://upload.wikimedia.org/wikipedia/commons/9/91/Octicons-mark-github.svg'
		}
	]
	beforeEach(() => {
		wrapper = shallow(<OrganizationList organizations={mockProps} />)
	});

	it('Renders correctly', () => {
		const tree = renderer.create(<Router><OrganizationList organizations={mockProps} /></Router>).toJSON();
		expect(tree).toMatchSnapshot();
	});
})