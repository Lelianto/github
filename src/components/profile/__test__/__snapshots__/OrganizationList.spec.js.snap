// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Organization List: Check content Renders correctly 1`] = `
Array [
  <a
    href="/org/Organization1"
    onClick={[Function]}
  >
    <img
      alt=""
      className="mr-1 mb-1"
      height="32px"
      size="32px"
      src="https://upload.wikimedia.org/wikipedia/commons/9/91/Octicons-mark-github.svg"
      width="32px"
    />
  </a>,
  <a
    href="/org/Organization2"
    onClick={[Function]}
  >
    <img
      alt=""
      className="mr-1 mb-1"
      height="32px"
      size="32px"
      src="https://upload.wikimedia.org/wikipedia/commons/9/91/Octicons-mark-github.svg"
      width="32px"
    />
  </a>,
]
`;
